creator-node/src/fileManager.js

```diff
@@ -95,18 +95,35 @@ async function saveFileToIPFSFromFS (req, srcPath) {
  *                  eg original.jpg or 150x150.jpg
  */
 async function saveFileForMultihashToFS (req, multihash, expectedStoragePath, gatewaysToTry, fileNameForImage = null) {
+  // stores all the stages of this function along with associated information relevant to that step
+  // in the try catch below, if any of the nested try/catches throw, it will be caught by the top level try/catch
+  // so we only need to print it once in the global catch or after everthing finishes except for any return statements
+  let decisionTree = []
+
   try {
     // will be modified to directory compatible route later if directory
     // TODO - don't concat url's by hand like this, use module like urljoin
     let gatewayUrlsMapped = gatewaysToTry.map(endpoint => `${endpoint.replace(/\/$/, '')}/ipfs/${multihash}`)
 
     const parsedStoragePath = path.parse(expectedStoragePath).dir
 
+    decisionTree.push({ stage: 'About to start running saveFileForMultihashToFS',
+      val: {
+        multihash,
+        gatewaysToTry,
+        gatewayUrlsMapped,
+        expectedStoragePath,
+        parsedStoragePath
+      },
+      time: Date.now()
+    })
     try {
       // calling this on an existing directory doesn't overwrite the existing data or throw an error
       // the mkdir recursive is equivalent to `mkdir -p`
       await mkdir(parsedStoragePath, { recursive: true })
+      decisionTree.push({ stage: 'Successfully called mkdir on local file system', val: parsedStoragePath, time: Date.now() })
     } catch (e) {
+      decisionTree.push({ stage: 'Error calling mkdir on local file system', val: parsedStoragePath, time: Date.now() })
```

Comment:
here and above you use `val` but all below calls use `vals`. i know it dosn't matter since they're never accessed by name anywhere but lets just fix

Reply:
fixed